builtin/aws/ecs/platform.go

```diff
@@ -286,6 +295,10 @@ func (p *Platform) resourceManager(log hclog.Logger, dcr *component.DeclaredReso
 // DeploymentId is a unique ID to be consistently used throughout our deployment
 type DeploymentId string
 
+// WorkspaceDestroy marks that we're destroying the entire workspace (not just the deployment), and
+// we should destroy global resources too
+type WorkspaceDestroy bool
```

Comment:
I'm a little unclear why we define this as a type...

Reply:
Ah good point - I added a comment clarifying: https://github.com/hashicorp/waypoint/pull/2684/files#diff-20946acdf4caf300d4d679801d8116bde77354aae07f8a988943474ca4351e85R295-R297

They're defined as types so that argmapper inside resource manager can wire them from where we invoke the overall destroy function ([here](https://github.com/hashicorp/waypoint/pull/2684/files#diff-20946acdf4caf300d4d679801d8116bde77354aae07f8a988943474ca4351e85R519)) to where we use them in individual resource funcs (like [here](https://github.com/hashicorp/waypoint/pull/2684/files#diff-20946acdf4caf300d4d679801d8116bde77354aae07f8a988943474ca4351e85R1570)).

I think it would be possible to use argmapper [named parameter matching](https://github.com/hashicorp/go-argmapper#named-and-typed-values) instead, but this felt better to me.